pyaerocom/griddeddata.py

```diff
@@ -684,12 +685,18 @@ def _get_info_from_filenames(self):
 
     def convert_unit(self, new_unit):
         """Convert unit of data to new unit"""
-# =============================================================================
-#         if self._size_GB > self._MAX_SIZE_GB:
-#             raise MemoryError('Cannot convert unit in {} since data is too '
-#                               'large ({} GB)'.format(self.name, self._size_GB))
-# =============================================================================
-        self.grid.convert_units(new_unit)
+        from_unit = self.units
+        var_name = self.var_name
+        import pyaerocom.units_helpers as uh
+        if self.var_name in uh.NON_SI_SUPPORTED.keys():
+            if uh.NON_SI_SUPPORTED[var_name] == new_unit:
+                _, fac = uh.unit_conversion_non_si_fac(var_name, from_unit)
+                cube = self.cube * fac
+                cube.units = new_unit
+                cube.var_name = var_name
+                self.cube = cube
```

Comment:
This kind of works for allowing conversion to non SI units.
unit_conversion_non_si_fac is just a copy of unit_conversion_custom_fac..

Reply:
But not sure how it will fit into the flow we need @jgliss.
We could check NON_SI_SUPPORTED in readgridded, and convert the units there, but that is not something we always want to do..